mautic/mautic
# Form field automatically remapping after being unmapped

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [✅]
| New feature/enhancement? (use the a.x branch)      | [❌]
| Deprecations?                          | [❌]
| BC breaks? (use the c.x branch)        | [❌]
| Automated tests included?              | [✅] 
| Related user documentation PR URL      |  
| Related developer documentation PR URL |  
| Issue(s) addressed                     |  



#### Description:

This PR fixes the following bugs:

1. Icons for mapped form fields are not displayed in the form field listing.
    

https://github.com/mautic/mautic/assets/6388925/fde7a4af-7dc6-419b-8a2c-6fa3fb9c4924

2. A non-mapped existing form field is auto-mapped when being edited.


https://github.com/mautic/mautic/assets/6388925/8c58afbb-6976-44ce-81ea-ec531609e54a



#### Steps to test this PR:


1. Icons for mapped form fields are not displayed in the form field listing.
    1. When adding a new mapped form field the icon indicating mapping should be visible right after clicking the "Add" button. See the first screen recording in the description.
2.
    1. Create a form with at least one non-mapped field of type "Email" and click “Save & Close”.
    2. Go back to the form edit page and refresh your browser.
    3. Click the "Pencil" icon to update the form field you created at 1. step.
    4. Navigate to the "Mapped Field" tab. The field should not be mapped to a contact's email. See the second screen recording in the description.
